File: Fork/app/src/main/java/com/eshed/fork/data/model/Direction.java
package com.eshed.fork.data.model;

public class Direction {
    int directionNumber;
    String directionText;
    int recipeID;

    public Direction(int directionNumber, String directionText, int recipeID) {
        this.directionNumber = directionNumber;
        this.directionText = directionText;
        this.recipeID = recipeID;
    }

    // temp
    public Direction(int directionNumber, String directionText) {
        this(directionNumber, directionText, -1);
    }

    public int getDirectionNumber() {
        return directionNumber;
    }

    public String getDirectionText() {
        return directionText;
    }
}

File: Fork/app/src/main/java/com/eshed/fork/Data/model/Direction.kt
package com.eshed.fork.Data.model

import com.google.firebase.database.Exclude

data class Direction
@JvmOverloads constructor(
    var directionNumber: Int = -1,
    var directionText: String = "",
    var recipeID: Int = -1
    ) {

    @Exclude
    override fun toString(): String {
        return "$directionNumber . $directionText"
    }

    @Exclude
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "directionNumber" to directionNumber,
            "directionText" to directionText,
            "recipeID" to recipeID
        )
    }
}

